apps/system/test_softlockup/test_softlockup.c

```diff
@@ -0,0 +1,189 @@
+{
+	for (;;) {
+		/* Do Nothing and Hog */
+	}
+	return NULL;
+}
+
+/****************************************************************************
+ * Name: thread_function
+ *
+ * Description:
+ *   This function terminates after a finite period.
+ *
+ * Input parameters:
+ *   void
+ *
+ * Returned Value:
+ *   void
+ *
+ ****************************************************************************/
+
+static void *thread_function(void *parameter)
+{
+	int i;
+	for (i = 0; i < 5; i++) {
+		sleep(1);
+	}
+	return NULL;
+}
+
+/****************************************************************************
+ * Name: test_low_prio_hog_thread
+ *
+ * Description:
+ *   This function creates two pthreads. One pthread is of high priority and
+ *   terminates after a finite period and second pthread runs in an infinite
+ *   loop.
+ *
+ * Input parameters:
+ *   void
+ *
+ * Returned Value:
+ *   void
+ *
+ ****************************************************************************/
+
+void test_low_prio_hog_thread(void)
+{
+	pthread_t thread1;
+	pthread_t thread2;
+	pthread_attr_t attr1;
+	pthread_attr_t attr2;
+	struct sched_param sparam;
+
+	(void)pthread_attr_init(&attr1);
+	sparam.sched_priority = 225;
+	(void)pthread_attr_setschedparam(&attr1, & sparam);
+	(void)pthread_create(&thread1, &attr1, thread_function, NULL);
+	pthread_setname_np(thread1, "hp_thread");
+
+	(void)pthread_attr_init(&attr2);
+	sparam.sched_priority = 150;
+	(void)pthread_attr_setschedparam(&attr2, & sparam);
+	(void)pthread_create(&thread2, &attr2, thread_hogging_function, NULL);
+	pthread_setname_np(thread2, "lp_hog_thread");
+
+}
+
+/****************************************************************************
+ * Name: test_normal_prio_hog_thread
+ *
+ * Description:
+ *   This function creates a single pthread which is runing in an infinite
+ *   loop.
+ *
+ * Input parameters:
+ *   void
+ *
+ * Returned Value:
+ *   void
+ *
+ ****************************************************************************/
+
+void test_normal_prio_hog_thread(void)
+{
+	pthread_t thread1;
+	pthread_attr_t attr;
+	struct sched_param sparam;
+
+	(void)pthread_attr_init(&attr);
+	sparam.sched_priority = 150;
+	(void)pthread_attr_setschedparam(&attr, & sparam);
+	(void)pthread_create(&thread1, &attr, thread_hogging_function, NULL);
+	pthread_setname_np(thread1, "hog_thread");
+
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int test_softlockup_main(int argc, char **argv)
+#endif
+{
+	char ch;
+	printf("\nPress A - For Normal Priority Hogging Thread Testcase\n");
+	printf("Press B - For Low Priority Hogging Thread Testcase\n");
+
+	ch = getchar();
+	printf("%c", ch);
+	if (ch == 'A' || ch == 'a') {
+		printf("\nStarting normal priority hogging thread testcase\n");
+		sleep(1);
```

Comment:
sleep: need to check a conditional, disable_signal.
pthread : disable_pthread.

Reply:
!DISABLE_SIGNALS and !DISABLE_PTHREAD have been checked in the Kconfig for this feature to be used

Reply:
There is only checking PTHREAD. no SIGNAL.

Reply:
@sunghan-chang , actually when this test case is selected, it will select the CONFIG_SOFT_LOCKUP_DETECT  kernel feature flag . This internally checks for !DISABLE_SIGNALS. I didn't add that check to avoid  redundancy